scripts/gen_api_vimdoc.py

```diff
@@ -471,67 +492,100 @@ def gen_docs(config):
                 if doc:
                     filename = os.path.basename(filename)
                     name = section_name.get(filename, name)
-                    title = '%s Functions' % name
-                    helptag = '*api-%s*' % name.lower()
+                    
+                    if filename.endswith('.lua'):
+                        title = '%s Lua Functions' % name
+                        helptag = '*lua-%s*' % name.lower()
+                    else:
+                        title = '%s Functions' % name
+                        helptag = '*api-%s*' % name.lower()
                     sections[filename] = (title, helptag, doc)
 
     if not sections:
         return
 
-    docs = ''
+    apidocs = ''
+    luadocs = ''
 
     i = 0
-    for filename in section_order:
+    for filename in api_section_order:
         if filename not in sections:
             continue
         title, helptag, section_doc = sections.pop(filename)
 
         i += 1
-        docs += sep
-        docs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
-        docs += section_doc
-        docs += '\n\n\n'
+        apidocs += sep
+        apidocs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
+        apidocs += section_doc
+        apidocs += '\n\n\n'
+
+    for filename in luadoc_section_order:
```

Comment:
Rather than trying to do everything in one pass, can we avoid this extra/duplicate code by choosing the "mode" at invocation time? Then this will be something like:

    for filename in CONFIG[mode]["section_order"]: